auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -704,4 +750,101 @@ private static void buildParameterizedTypeArguments(CodeBlock.Builder block, Typ
       block.add("$T.class", typeArg);
     }
   }
+
+  private static Optional<ExecutableElement> findDefaultBuilder(TypeElement autoValueClass, TypeElement builderClass) {
+    return ElementFilter.methodsIn(autoValueClass.getEnclosedElements())
+        .stream()
+        .filter(e -> MoreElements.isAnnotationPresent(e, GsonDefaultBuilder.class))
+        .filter(MoreElements.hasModifiers(Modifier.STATIC)::apply)
+        .filter(e -> MoreTypes.equivalence().equivalent(e.getReturnType(), builderClass.asType()))
+        .filter(e -> e.getParameters().isEmpty())
+        .findFirst();
+  }
+
+  /**
+   * Searches the builder for an method has no parameters, returns the autovalue class, is not
+   * private, and is not static. If there is a valid method annotated with {@link GsonBuild}, return
+   * that method. Otherwise, return the method named "build" if present.
+   */
+  private static Optional<ExecutableElement> findBuildMethod(
+      TypeElement autoValueClass, TypeElement builderElement) {
+    List<ExecutableElement> methods = ElementFilter.methodsIn(builderElement.getEnclosedElements())
+        .stream()
+        .filter(e -> !MoreElements.hasModifiers(STATIC, PRIVATE).apply(e))
+        .filter(e -> e.getParameters().isEmpty())
+        .filter(e -> MoreTypes.equivalence().equivalent(e.getReturnType(), autoValueClass.asType()))
+        .collect(Collectors.toList());
+
+    if (methods.isEmpty()) {
+      return Optional.empty();
+    }
+
+    List<ExecutableElement> annotatedMethods = methods.stream()
+        .filter(e -> MoreElements.isAnnotationPresent(e, GsonBuild.class))
+        .collect(Collectors.toList());
+
+    if (annotatedMethods.size() > 1) {
+      return Optional.empty();
+    } else if (annotatedMethods.size() == 1) {
+      return Optional.of(Iterables.getOnlyElement(annotatedMethods));
+    }
+
+    return methods.stream()
+        .filter(e -> e.getSimpleName().contentEquals("build"))
+        .findFirst();
+  }
+
+  static boolean validateTypeAdapter(Context context) {
```

Comment:
since this returns a boolean, I think it should be named with the `is` prefix. The verb tense makes me think it throws if it's not, but maybe that's just me